internal/controllers/user_devices_controller.go

```diff
@@ -416,37 +416,47 @@ func (udc *UserDevicesController) GetUserDevices(c *fiber.Ctx) error {
 		return helpers.ErrorResponseHandler(c, err, fiber.StatusInternalServerError)
 	}
 
-	if !udc.Settings.IsProduction() {
-		toCheck := make(map[uint32]*models.UserDeviceAPIIntegration)
+	{
+		const sourcePrefix = "dimo/integration/"
+		type checkKey struct {
+			TokenID       uint32
+			IntegrationID string
+		}
+		toCheck := make(map[checkKey]*models.UserDeviceAPIIntegration)
 		for _, ud := range devices {
 			if ud.TokenID.IsZero() {
 				continue
 			}
 			for _, udai := range ud.R.UserDeviceAPIIntegrations {
-				if udai.IntegrationID == "2lcaMFuCO0HJIUfdq8o780Kx5n3" {
-					if udai.Status != "Active" {
-						tok, _ := ud.TokenID.Uint64()
-						toCheck[uint32(tok)] = udai
-					}
-					break
+				// TODO(elffjs): Really no point in doing this for synthetics if the job hasn't started.
+				// Hard to tell this at this point
+				if udai.Status != "Active" {
+					tok, _ := ud.TokenID.Uint64()
+					toCheck[checkKey{uint32(tok), udai.IntegrationID}] = udai
 				}
+				break
 			}
 		}
 
 		if len(toCheck) != 0 {
 			udc.log.Info().Str("userId", userID).Msgf("Checking %d Ruptela connections.", len(toCheck))
```

Comment:
Should we get rid from Ruptela word, as I understand these changes are generic?

Reply:
Not done

Reply:
Set now